c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,101 @@
-﻿using Newtonsoft.Json;
+﻿using JsonDiffPatchDotNet;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using System;
 using System.Collections;
 using System.Collections.Generic;
 
 namespace c_sharp_json_diff
 {
-    public class Result {
-        public List<Diff> Diffs { get; set; } = new List<Diff>();
-
-    }
-    public class Diff
-    {
-        [JsonProperty("property")]
-        public string jsonProperty { get; set; } = "";
-        [JsonProperty("previous")]
-        public double previous { get; set; }
-        [JsonProperty("current")]
-        public double current { get; set; }
-        [JsonProperty("difference")]
-        public double difference { get; set; }
-    }
     public class JsonDiff
     {
-        public static Result GetDiff(IDictionary<string, double> first, IDictionary<string, double> second)
+        private string _leftJsonString = "";
+        private string _rightJsonString = "";
+
+        public JsonDiff(string left, string right)
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
+            _leftJsonString = left;
+            _rightJsonString = right;
+        }
+
+        public JObject NumericDiff(bool isAbsoluteDiff)
+        {
+            string diff = "";
+            try
             {
-                if (firstKeys.Contains(key))
+                diff = JsonDiffBetweenTwo(_leftJsonString, _rightJsonString);
+                if (diff == null)
                 {
-                    matches.Add(key);
+                    return JObject.Parse(@"{}");
                 }
+                JObject diffJsonObject = JObject.Parse(diff);
+                JObject diffReturnObject = CalculateNumericDiff(diffJsonObject, isAbsoluteDiff);
+                return diffReturnObject;
+            } catch(Exception e)
+            {
+                return JObject.Parse(diff + e.StackTrace);
             }
-            foreach(string key in matches)
+        }
+
+        public string NumericDiffAsString(bool isAbsoluteDiff)
+        {
+            JObject diff = NumericDiff(isAbsoluteDiff);
+            return diff.ToString(Formatting.Indented);
+
+        }
+
+        private JObject CalculateNumericDiff(JObject jObject, bool isAbsoluteDiff)
+        {
+            foreach (KeyValuePair<string, JToken?> prop in jObject)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                if(prop.Key == "_t")
```

Comment:
What is the meaning of `_t`? If it is important, add a constant for it.

Reply:
This is an additional property being added by the external JsonDiffPatch library when the Json is an array and has to be ignored. I tried understanding why this was being added, but I couldn't quite understand the reason in the library source code